ldap/servers/slapd/passwd_extop.c

```diff
@@ -132,6 +133,14 @@ passwd_apply_mods(Slapi_PBlock *pb_orig, const Slapi_DN *sdn, Slapi_Mods *mods,
         }
 
         Slapi_PBlock *pb = slapi_pblock_new();
+        pwpolicy = new_passwdPolicy(pb_orig, slapi_sdn_get_ndn(sdn));
+        if (pw_is_pwp_admin(pb_orig, pwpolicy, PWP_ADMIN_OR_ROOTDN)) {
+            /* If this is root DN or password admin set is_requestor_root so
+             * set this in the new pblock so password updates are correctly
+             * applied */
+            int is_requestor_root = 1;
+            slapi_pblock_set(pb, SLAPI_REQUESTOR_ISROOT, &is_requestor_root);
```

Comment:
Weird, IIUC pw_is_pwp_admin return True if pblock[SLAPI_REQUESTOR_ISROOT] is already 1. Is that setting required ?

Reply:
> Weird, IIUC pw_is_pwp_admin return True if pblock[SLAPI_REQUESTOR_ISROOT] is already 1. Is that setting required ?

It is the easiest way to set the "new" pblock used for the mod that updates the password.  So we check the "current/old" pblock to see if it's a password admin, if it is then we set the SLAPI_REQUESTOR_ISROOT to 1 in the "new" pblock so the password is updated as expected

Reply:
oppsss.. my bad, I read too fast and missed there were two pblocks. Ignore my comment